tests/Unit/Elliptic/Actions/Test_InitializeSystem.cpp

```diff
@@ -0,0 +1,249 @@
+                                              element_id);
+    const auto get_tag = [&runner, &element_id](auto tag_v) -> decltype(auto) {
+      using tag = std::decay_t<decltype(tag_v)>;
+      return ActionTesting::get_databox_tag<element_array, tag>(runner,
+                                                                element_id);
+    };
+
+    check_compute_items(runner, element_id);
+
+    CHECK(get_tag(ScalarFieldTag{}) == Scalar<DataVector>{{{{4, 0.}}}});
+    CHECK(get_tag(LinearSolver::Tags::OperatorAppliedTo<ScalarFieldTag>{}) ==
+          Scalar<DataVector>{{{{4, 0.}}}});
+    CHECK(get(get_tag(Tags::Source<ScalarFieldTag>{})) ==
+          get<0>(get_tag(Tags::Coordinates<1, Frame::Inertial>{})));
+    CHECK(get(get_tag(SomePrefix<ScalarFieldTag>{})).size() == 4);
+    CHECK(get(get_tag(LinearSolver::Tags::OperatorAppliedTo<
+                      SomePrefix<ScalarFieldTag>>{}))
+              .size() == 4);
+  }
+  {
+    INFO("2D");
+    // Which element we work with does not matter for this test
+    const ElementId<2> element_id{0, {{SegmentId{2, 1}, SegmentId{0, 0}}}};
+    const domain::creators::Rectangle<Frame::Inertial> domain_creator{
+        {{-0.5, 0.}}, {{1.5, 2.}}, {{false, false}}, {{2, 0}}, {{4, 3}}};
+    // Register the coordinate map for serialization
+    PUPable_reg(
+        SINGLE_ARG(domain::CoordinateMap<Frame::Logical, Frame::Inertial,
+                                         domain::CoordinateMaps::ProductOf2Maps<
+                                             domain::CoordinateMaps::Affine,
+                                             domain::CoordinateMaps::Affine>>));
+
+    using metavariables = Metavariables<2>;
+    using element_array = ElementArray<2, metavariables>;
+    ActionTesting::MockRuntimeSystem<metavariables> runner{
+        {AnalyticSolution<2>{}}};
+    ActionTesting::emplace_component_and_initialize<element_array>(
+        &runner, element_id,
+        {domain_creator.create_domain(), domain_creator.initial_extents()});
+    ActionTesting::next_action<element_array>(make_not_null(&runner),
+                                              element_id);
+    runner.set_phase(metavariables::Phase::Testing);
+    ActionTesting::next_action<element_array>(make_not_null(&runner),
+                                              element_id);
+    const auto get_tag = [&runner, &element_id](auto tag_v) -> decltype(auto) {
+      using tag = std::decay_t<decltype(tag_v)>;
+      return ActionTesting::get_databox_tag<element_array, tag>(runner,
+                                                                element_id);
+    };
+
+    check_compute_items(runner, element_id);
+
+    CHECK(get_tag(ScalarFieldTag{}) == Scalar<DataVector>{{{{12, 0.}}}});
+    CHECK(get_tag(LinearSolver::Tags::OperatorAppliedTo<ScalarFieldTag>{}) ==
+          Scalar<DataVector>{{{{12, 0.}}}});
+    CHECK(get(get_tag(Tags::Source<ScalarFieldTag>{})) ==
```

Comment:
it might be helpful to have an inline comment explaining that this check is against the analytic solution above (otherwise potentially confusing why the initialization gives these values) - same for different dimensionalities

Reply:
Will add!

Reply:
More generally, some comments about what the tests are testing would be helpful here

Reply:
Squashed in some more comments in the test